src/app/tests/suites/TestIcdManagementCluster.yaml

```diff
@@ -0,0 +1,291 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: ICD Management Cluster Tests
+
+config:
+    nodeId: 0x12344321
+    cluster: "ICD Management"
+    endpoint: 0
+
+tests:
+    - label: "Wait for the commissioned device to be retrieved"
+      cluster: "DelayCommands"
+      command: "WaitForCommissionee"
+      arguments:
+          values:
+              - name: "nodeId"
+                value: nodeId
+
+    - label: "Read IdleModeInterval"
+      command: "readAttribute"
+      attribute: "IdleModeInterval"
+      response:
+          value: 500
+
+    - label: "Read ActiveModeInterval"
+      command: "readAttribute"
+      attribute: "ActiveModeInterval"
+      response:
+          value: 300
+
+    - label: "Read ActiveModeThreshold"
+      command: "readAttribute"
+      attribute: "ActiveModeThreshold"
+      response:
+          value: 300
+
+    - label: "Read ICDCounter"
+      command: "readAttribute"
+      attribute: "ICDCounter"
+      response:
+          value: 0
+
+    - label: "Read ClientsSupportedPerFabric"
+      command: "readAttribute"
+      attribute: "ClientsSupportedPerFabric"
+      response:
+          value: 2
+
+    - label: "Unregister 1.0"
+      command: "UnregisterClient"
+      arguments:
+          values:
+              - name: "CheckInNodeID"
+                value: 101
+      response:
+          error: NOT_FOUND
+
+    - label: "Unregister 2.0"
+      command: "UnregisterClient"
+      arguments:
+          values:
+              - name: "CheckInNodeID"
+                value: 102
+      response:
+          error: NOT_FOUND
+
+    - label: "Unregister 3.0"
+      command: "UnregisterClient"
+      arguments:
+          values:
+              - name: "CheckInNodeID"
+                value: 102
+      response:
+          error: NOT_FOUND
+
+    - label: "Read RegisteredClients"
+      command: "readAttribute"
+      attribute: "RegisteredClients"
+      response:
+          value: []
+
+    - label: "Register 1.0 (invalid key length)"
+      command: "RegisterClient"
+      arguments:
+          values:
+              - name: "CheckInNodeID"
+                value: 101
+              - name: "MonitoredSubject"
+                value: 1001
+              - name: "Key"
+                value: "\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f\xff"
+      response:
+          error: FAILURE
```

Comment:
Since this is a constraint violation, I would expect a different error here, unless the spec says to use FAILURE.  Does it?

Also, please test a too-short key.

Reply:
I didn't find anything in the spec that says what error should be returned, nevertheless I changed it to CONSTRAINT_ERROR.

I also didn't find any minimum size constraint. As far as I know, a zero-length key is valid.

Reply:
I assumed that with the generation, a check size was automatically done when the size is fixed.
Is that not the case?

Reply:
> I didn't find anything in the spec that says what error should be returned

See "8.8.2.3. Incoming Invoke Request Action" under "Invoke Execution" where it says:

> If a data field violates expected constraints, a CommandStatusIB SHOULD be generated with an error status of CONSTRAINT_ERROR.

It's a "SHOULD" because cluster specifications can override.

> I assumed that with the generation, a check size was automatically done when the size is fixed.

Can't be done, because per spec cluster specifications can override the behavior when the constraint is violated.  So the generic generated machinery does not have enough information to do the check and return the right error code.